main/src/main/scala/org/clulab/processors/clu/Veil.scala

```diff
@@ -0,0 +1,244 @@
+
+    def unveilStringArray(veiledArrayOpt: Option[Array[String]], veil: String): Option[Array[String]] =
+        this.unveilStringArray(veiledArrayOpt, sentenceIndex, veil)
+
+    unveiledSentence.tags     = unveilStringArray(unveiledSentence.tags,     Veil.veiledTag)
+    unveiledSentence.lemmas   = unveilStringArray(unveiledSentence.lemmas,   Veil.veiledLemma)
+    unveiledSentence.entities = unveilStringArray(unveiledSentence.entities, Veil.veiledEntity)
+    unveiledSentence.norms    = unveilStringArray(unveiledSentence.norms,    Veil.veiledNorm)
+    unveiledSentence.chunks   = unveilStringArray(unveiledSentence.chunks,   Veil.veiledChunk)
+
+    unveiledSentence.syntacticTree = unveilSyntacticTree(unveiledSentence.syntacticTree)
+    unveiledSentence.graphs = unveilGraphs(unveiledSentence.graphs, sentenceIndex)
+    unveiledSentence.relations = unveilRelations(unveiledSentence.relations)
+    unveiledSentence
+  }
+
+  protected def unveilDocument(veiledDocument: Document): Document = {
+    val unveiledSentences = veiledDocument.sentences.zipWithIndex.map { case (veiledSentence, sentenceIndex) =>
+      unveilSentence(veiledSentence, sentenceIndex)
+    }
+    val unveiledAnnotatedDocument = veiledDocument.copy(unveiledSentences)
+
+    unveiledAnnotatedDocument
+  }
+
+  def annotate(processor: Processor): Document = {
+    val veiledAnnotatedDocument = processor.annotate(veiledDocument)
+    val unveiledAnnotatedDocument = unveilDocument(veiledAnnotatedDocument)
+
+    unveiledAnnotatedDocument
+  }
+}
+
+/** Demonstrate how either parts of the text or Document can be veiled.
+  */
+object VeilApp extends App {
+
+  /** Treat this text as if the letters "(Hahn-Powell, 2012)" did not exist
+    *  for the purpose of mkDocument, but do include them in the text.
+    */
+  def veilText(processsor: Processor): Unit = {
+    val text = "To be loved by unicorns is the greatest gift of all (Hahn-Powell, 2012)."
+    val veiledLetters = Seq(Range.inclusive(text.indexOf('('), text.indexOf(')')))
+    val veiledText = new VeiledText(text, veiledLetters)
+    val document = veiledText.mkDocument(processor)
+
+    Using.resource(new PrintWriter("veiledLetters.out")) { printWriter =>
+      val documentSerializer = new DocumentSerializer()
+
+      documentSerializer.save(document, printWriter)
+    }
+  }
+
+  /** Treat this text as if the words "( Hahn-Powell , 2012 )" did not exist
+    * for the purpose of annotate, but do include them in the document.
+    */
+  def veilDocument(processor: Processor): Unit = {
+    val text = "To be loved by unicorns is the greatest gift of all (Hahn-Powell, 2012)."
+    val document = processor.mkDocument(text)
+    val veiledWords = Seq((0, Range.inclusive(document.sentences(0).raw.indexOf("("), document.sentences(0).raw.indexOf(")"))))
+    val veiledDocument = new VeiledDocument(document, veiledWords)
+    val annotatedDocument = veiledDocument.annotate(processor)
```

Comment:
Would be good to add a type hint here to make it clear that `annotatedDocument` is a `Document`.